web_pwa_oca/controllers/main.py

```diff
@@ -14,53 +26,58 @@ def get_asset_urls(self, asset_xml_id):
                 urls.append(asset[1]['src'])
         return urls
 
-    @route('/service-worker.js', type='http', auth="public")
-    def service_worker(self):
-        qweb = request.env['ir.qweb'].sudo()
+    def _get_pwa_params(self):
+        """ Get javascript PWA class initialzation params """
         urls = []
-        urls.extend(self.get_asset_urls("web.assets_common"))
-        urls.extend(self.get_asset_urls("web.assets_backend"))
+        urls.extend(self._get_asset_urls("web.assets_common"))
+        urls.extend(self._get_asset_urls("web.assets_backend"))
         version_list = []
         for url in urls:
             version_list.append(url.split('/')[3])
         cache_version = '-'.join(version_list)
-        mimetype = 'text/javascript;charset=utf-8'
-        content = qweb.render('web_pwa_oca.service_worker', {
-            'pwa_cache_name': cache_version,
-            'pwa_files_to_cache': urls,
-        })
-        return request.make_response(content, [('Content-Type', mimetype)])
+        return [
+            cache_version,
+            urls
+        ]
+
+    @route('/service-worker.js', type='http', auth="public")
+    def render_service_worker(self):
+        """ Route to register the service worker in the 'main' scope ('/') """
+        return request.render('web_pwa_oca.service_worker', {
+            'pwa_scripts': self._get_pwa_scripts(),
+            'pwa_params': self._get_pwa_params(),
+        }, headers=[('Content-Type', 'text/javascript;charset=utf-8')])
 
     @route('/web_pwa_oca/manifest.json', type='http', auth="public")
```

Comment:
I explain this in some comments below:

```suggestion
    @route('/web_pwa_oca/manifest.json', type='json', auth="public")
```

Reply:
Need be 'http' type because is loaded through a "link" html tag that launch GET petition. But i moved all the manifest definition to python.